Homeworks/Homework SQL/SQLQuery-Homework.sql

```diff
@@ -0,0 +1,339 @@
+ALTER TABLE Books
+DROP COLUMN AuthorFirstName;
+ALTER TABLE Books
+DROP COLUMN AuthorLastName;
+ALTER TABLE Books
+DROP COLUMN Type;
+
+CREATE TABLE Author
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+FirstName VARCHAR (50),
+LastName VARCHAR (50),
+Email VARCHAR (30),
+)
+
+CREATE TABLE TYPE
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+Name VARCHAR (30),
+)
+
+ALTER TABLE Books
+ADD AuthorId INT REFERENCES Author(id);
+
+ALTER TABLE Books
+ADD TypeId INT REFERENCES Type(id);
+
+SELECT *
+FROM Books;
+
+ALTER TABLE Librarians
+DROP COLUMN IsOnHoliday;
+
+CREATE TABLE IsOnHoliday
+(
+Id INT IDENTITY(1,1) PRIMARY KEY,
+HolidayStatus VARCHAR (20),
+)
+
+ALTER TABLE Librarians
+ADD IsOnHolidayId INT REFERENCES IsOnHoliday(id);
+
+INSERT INTO Author
+VALUES ('ana', 'mitrea', 'am@gmail.com');
+INSERT INTO Author
+VALUES ('andrei', 'stan', 'as@gmail.com');
+INSERT INTO Author
+VALUES ('diana', 'ciocan', 'ac@gmail.com');
+INSERT INTO Author
+VALUES ('teodor', 'nistor', 'tn@gmail.com');
+INSERT INTO Author
+VALUES ('stefania', 'iancu', 'si@gmail.com');
+
+SELECT *
+FROM Author;
+
+--crime, fantasy, horror, adventure
+INSERT INTO TYPE
+VALUES ('crime');
+INSERT INTO TYPE
+VALUES ('fantasy');
+INSERT INTO TYPE
+VALUES ('horror');
+INSERT INTO TYPE
+VALUES ('adventure');
+
+SELECT *
+FROM TYPE;
+
+INSERT INTO Books
+VALUES ('Book1',1,1);
+INSERT INTO Books
+VALUES ('Book2',1,2);
+INSERT INTO Books
+VALUES ('Book3',2,3);
+INSERT INTO Books
+VALUES ('Book4',3,4);
+INSERT INTO Books
+VALUES ('Book5',4,2);
+INSERT INTO Books
+VALUES ('Book6',5,4);
+INSERT INTO Books
+VALUES ('Book7',4,3);
+INSERT INTO Books
+VALUES ('Book8',3,2);
+INSERT INTO Books
+VALUES ('Book9',2,1);
+
+SELECT *
+FROM Books;
+
+INSERT INTO Libraries
+VALUES ('Library 1','Bv. Independetei','Iasi');
+INSERT INTO Libraries
+VALUES ('Library 2','Bv. Eroilor','Bucuresti');
+INSERT INTO Libraries
+VALUES ('Library 3','Strada Brancusi','Brasov');
+INSERT INTO Libraries
+VALUES ('Library 4','Bv. Unirii','Bucuresti');
+INSERT INTO Libraries
+VALUES ('Library 5','Strada Stefan Cel Mare','Iasi');
+INSERT INTO Libraries
+VALUES ('Library 6','Bv. Socola','Iasi');
+
+-- Forgot to change library number
+DELETE FROM Libraries
+WHERE Id=6;
+
+SELECT * 
+FROM Libraries;
+
+INSERT INTO IsOnHoliday
+VALUES ('True');
+INSERT INTO IsOnHoliday
+VALUES ('False');
+
+SELECT *
+FROM IsOnHoliday;
+
+INSERT INTO Librarians
+VALUES ('Gustav','2022-01-01',1,1);
+INSERT INTO Librarians
+VALUES ('Robert','2010-01-01',7,2);
+INSERT INTO Librarians
+VALUES ('Giulia','2012-01-01',8,1);
+INSERT INTO Librarians
+VALUES ('Maxi','2019-01-01',9,2);
+INSERT INTO Librarians
+VALUES ('Vlad','2013-01-01',10,2);
+INSERT INTO Librarians
+VALUES ('Dana','2023-01-01',1,2);
+INSERT INTO Librarians
+VALUES ('Mihai','2007-01-01',11,1);
+INSERT INTO Librarians
+VALUES ('Ionut','2019-01-01',8,2);
+
+INSERT INTO Sales
```

Comment:
Same error as above on Sales. By running the script for the first time I have no LibraryId greater than 7. Also for LibrarianId there is no id 9, 10 and 11 considering that above you only inserted 8 librarians into the database.